### PR TITLE
Update Test for validating overriding of TPA load context assemblies

### PR DESCRIPTION
**Pull request by gkhanna79**
Also fixes https://github.com/dotnet/corefx/issues/8563

@jkotas PTAL

CC @stephentoub

### CODE REVIEW

File: src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs [hunk 1]
```diff
@@ -80,18 +80,19 @@ public static void LoadFromAssemblyName_AssemblyNotFound()
...
-            // or supply your own (via AssemblyLoadContext) for your own assemblies.
-            Assert.Throws(typeof(FileLoadException), 
-                () => loadContext.LoadFromAssemblyName(asmName));
+            // We should be able to override (and thus, load) assemblies that were
+            // loaded in TPA load context.
+            var asm = loadContext.LoadFromAssemblyName(asmName);
+            Assert.NotNull(asm);
+            var loadedContext = AssemblyLoadContext.GetLoadContext(asm);
+            Assert.NotNull(loadedContext);
+            Assert.Equal(loadContext, loadedContext);
```
Comment (stephentoub) [inline src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]: Do we expect this to be the exact same object instance, or two different instances that have defined equality?  If the former, it'd be better to use Assert.Same.

Comment (gkhanna79) [inline src/System.Runtime.Loader/tests/AssemblyLoadContextTest.cs]: It is expect to be the same instance. I will fix this accordingly.

**Comment by jkotas** [comment]
LGTM

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

**Comment by gkhanna79** [comment]
@stephentoub Do you know if CoreFX got an updated CoreCLR packaging containing my ALC changes?

**Comment by stephentoub** [comment]
> @stephentoub Do you know if CoreFX got an updated CoreCLR packaging containing my ALC changes?

It did not.  Yesterday's CoreCLR package was broken due to the lack of the JIT binary.  We're still waiting on today's.

**Comment by gkhanna79** [comment]
I thought you mentioned that non-Windows build got my change. Is that not correct? Are they still using older CoreCLR?

**Comment by stephentoub** [comment]
> I thought you mentioned that non-Windows build got my change

Some did, some didn't.  What I said was "that not all of our CoreFx legs pull CoreCLR from packages; some pull the latest successful build from the CoreCLR repo."  That's true.  Some of these Unix legs pull the latest CoreCLR artifacts from Jenkins (e.g. CentOS7.1), while some pull CoreCLR from the NuGet package.

**Comment by gkhanna79** [comment]
Ok - I guess that I need to wait another day.

**Comment by stephentoub** [comment]
Closing this; I had to cherry-pick the commit into https://github.com/dotnet/corefx/pull/8620 to get it to pass.

**Pull request closed without merging**